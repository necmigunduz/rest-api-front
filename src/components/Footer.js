import { logout } from '../sessions/sessions';
import Icon from './Icon';

const Footer = () => {
  const handleLogout = () => {
    logout();
  };

  return (
    <footer className="flex fixed footer">
      <Icon icon="fas fa-plus" path="/add" title="Add Measurement" />
      <Icon icon="fas fa-home" path="/" title="Home" />
      <Icon icon="fas fa-tasks" path="/progress" title="Progress" />
      <Icon icon="fas fa-sign-out-alt" path="/users/login" title="Logout" clickIconHandler={handleLogout} />
    </footer>
  );
};

export default Footer;
