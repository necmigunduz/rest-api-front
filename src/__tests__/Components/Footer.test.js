import renderer from 'react-test-renderer';
import Footer from '../../components/Footer';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: () => ({
    pathname: 'www.instance.com/test',
  }),
}));

describe('Footer', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Footer />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
