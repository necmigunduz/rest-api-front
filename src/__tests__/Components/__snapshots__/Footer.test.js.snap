// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<footer
  className="flex fixed footer"
>
  <a
    className="icon background-dark-blue color-white"
    href="/add"
    onClick={[Function]}
  >
    <div>
      <div
        className="fas fa-plus big"
      />
      <div
        className="small m-top-10"
      >
        Add Measurement
      </div>
    </div>
  </a>
  <a
    className="icon background-dark-blue color-white"
    href="/"
    onClick={[Function]}
  >
    <div>
      <div
        className="fas fa-home big"
      />
      <div
        className="small m-top-10"
      >
        Home
      </div>
    </div>
  </a>
  <a
    className="icon background-dark-blue color-white"
    href="/progress"
    onClick={[Function]}
  >
    <div>
      <div
        className="fas fa-tasks big"
      />
      <div
        className="small m-top-10"
      >
        Progress
      </div>
    </div>
  </a>
  <a
    className="icon background-dark-blue color-white"
    href="/users/login"
    onClick={[Function]}
  >
    <div>
      <div
        className="fas fa-sign-out-alt big"
      />
      <div
        className="small m-top-10"
      >
        Logout
      </div>
    </div>
  </a>
</footer>
`;
